CMakeLists.txt

```diff
@@ -552,6 +553,78 @@ if(BUILD_PYTHON)
     Python::Module
   )
   install(TARGETS nvfuser DESTINATION lib)
+
+  # ------------------------------------------------
+  # build nvfuser next python library
+  # ------------------------------------------------
+  # nvfuser next bindings API sources
+  set(NVFUSER_PYTHON_NEXT_SRCS)
+  list(APPEND NVFUSER_PYTHON_NEXT_SRCS
+    ${NVFUSER_PYTHON_NEXT_BINDINGS}/extension.cpp
+    ${NVFUSER_PYTHON_NEXT_BINDINGS}/bindings.cpp
+  )
+  add_library(nvf_py_next_internal OBJECT ${NVFUSER_PYTHON_NEXT_SRCS})
+
+  # NOTE: For any future extension, change PYTHON_NEXT_EXTENSION to another
+  # name other than EXTENSION_NAME.
+  target_compile_definitions(nvf_py_next_internal PRIVATE
+    "-DTORCH_CUDA_BUILD_MAIN_LIB"
+    "-DC10_BUILD_MAIN_LIB=1"
+    PYTHON_NEXT_EXTENSION=_C_NEXT
+  )
+
+  add_library(nvfuser_next MODULE $<TARGET_OBJECTS:nvf_py_next_internal>)
+  target_compile_definitions(nvfuser_next PRIVATE
+    "-DTORCH_CUDA_BUILD_MAIN_LIB"
+    "-DC10_BUILD_MAIN_LIB=1"
+    PYTHON_NEXT_EXTENSION=_C_NEXT
+  )
+
+  if(NOT MSVC)
+    target_compile_options(nvf_py_next_internal PRIVATE -Wall -Wno-unused-function)
+    target_compile_options(nvf_py_next_internal PRIVATE -Werror)
+    set(NVF_LIB_SUFFIX ".so")
+  else()
+    set(NVF_LIB_SUFFIX ".pyd")
+  endif()
+
+  set_target_properties(nvf_py_next_internal PROPERTIES
+    C_STANDARD ${NVFUSER_C_STANDARD}
+    CUDA_STANDARD ${NVFUSER_CUDA_STANDARD}
+    CXX_STANDARD ${NVFUSER_CPP_STANDARD}
+    CXX_STANDARD_REQUIRED ON
+    CXX_VISIBILITY_PRESET hidden
+    INSTALL_RPATH
+    "$ORIGIN/lib:$ORIGIN/../nvidia/cuda_runtime/lib:$ORIGIN/../nvidia/cuda_nvrtc/lib:$ORIGIN/../../nvidia/cuda_cupti/lib:$ORIGIN/../torch/lib"
+    POSITION_INDEPENDENT_CODE Yes
+    VISIBILITY_INLINES_HIDDEN Yes
+  )
+  set_target_properties(nvfuser_next PROPERTIES
+    C_STANDARD ${NVFUSER_C_STANDARD}
+    CUDA_STANDARD ${NVFUSER_CUDA_STANDARD}
+    CXX_STANDARD ${NVFUSER_CPP_STANDARD}
+    CXX_STANDARD_REQUIRED ON
+    CXX_VISIBILITY_PRESET hidden
+    INSTALL_RPATH
+    "$ORIGIN/lib:$ORIGIN/../nvidia/cuda_runtime/lib:$ORIGIN/../nvidia/cuda_nvrtc/lib:$ORIGIN/../../nvidia/cuda_cupti/lib:$ORIGIN/../torch/lib"
+    POSITION_INDEPENDENT_CODE Yes
+    SUFFIX ${NVF_LIB_SUFFIX}
+    VISIBILITY_INLINES_HIDDEN Yes
+  )
+
+  target_include_directories(nvf_py_next_internal PUBLIC ${NVFUSER_PYTHON_NEXT_BINDINGS})
+  target_link_libraries(nvf_py_next_internal PRIVATE
+    codegen_internal
+    "${TORCH_INSTALL_PREFIX}/lib/libtorch_python.so"
+    pybind11::pybind11 pybind11::headers
+  )
+
+  target_link_libraries(nvfuser_next PRIVATE
+    nvf_py_next_internal
+    codegen_internal
+    Python::Module
+  )
+  install(TARGETS nvfuser_next DESTINATION lib)
```

Comment:
my nitpick on this PR is that, we would want to avoid duplicate the code that's shared between the two pybind module.

I'm also leaning towards moving this down to the `./python/CMakeLists.txt`.
But we don't have to do that in this PR. We can leave that as a follow up clean PR.

Reply:
What do you want to move to `.python/CMakeLists.txt`?

Build engineer is on the case.

Reply:
just simply for logic separation.
If I'm adding another python module. I don't need to see the entire build files for nvfuser core lib.